src/openzaak/config/bootstrap/authorizations.py

```diff
@@ -0,0 +1,118 @@
+    logger.info("Removing all existing authorization configuration")
+
+    Applicatie.objects.all().delete()
+    Autorisatie.objects.all().delete()
+    CatalogusAutorisatie.objects.all().delete()
+    JWTSecret.objects.all().delete()
+
+
+class AuthorizationConfigurationStep(BaseConfigurationStep):
+    """
+    Configure the Applicaties and Autorisaties.
+
+    From: https://open-zaak.readthedocs.io/en/stable/manual/api-authorizations.html
+    """
+
+    verbose_name = "Authorization configuration"
+    required_settings = [
+        "AUTHORIZATIONS_CONFIG_FIXTURE_PATH",
+    ]
+    enable_setting = "AUTHORIZATIONS_CONFIG_ENABLE"
+
+    def is_configured(self) -> bool:
+        with open(settings.AUTHORIZATIONS_CONFIG_FIXTURE_PATH) as f:
+            data = yaml.safe_load(f.read())
+            existing_uuids = set(
+                str(uuid) for uuid in Applicatie.objects.values_list("uuid", flat=True)
+            )
+            applicatie_uuids = {
+                row["fields"]["uuid"]
+                for row in data
+                if row["model"] == "authorizations.applicatie"
+            }
+            applicaties_configured = applicatie_uuids.issubset(existing_uuids)
+
+            existing_identifiers = set(
+                str(uuid)
+                for uuid in JWTSecret.objects.values_list("identifier", flat=True)
+            )
+
+            identifiers = {
+                row["fields"]["identifier"]
+                for row in data
+                if row["model"] == "vng_api_common.jwtsecret"
+            }
+            secrets_configured = identifiers.issubset(existing_identifiers)
+
+        return applicaties_configured and secrets_configured
+
+    @transaction.atomic
+    def configure(self):
+        if settings.AUTHORIZATIONS_CONFIG_DELETE_EXISTING:
+            delete_existing_configuration()
+
+        with open(settings.AUTHORIZATIONS_CONFIG_FIXTURE_PATH) as original_file:
+            content = original_file.read()
+
+            if settings.AUTHORIZATIONS_CONFIG_DOMAIN_MAPPING_PATH:
+                with open(
+                    settings.AUTHORIZATIONS_CONFIG_DOMAIN_MAPPING_PATH
+                ) as mapping_file:
+                    mapping = yaml.safe_load(mapping_file)
+                    for entry in mapping:
+                        for env, domain in entry.items():
+                            if env == settings.ENVIRONMENT:
+                                continue
+
+                            content = content.replace(
+                                domain, entry[settings.ENVIRONMENT]
+                            )
+
+            # Load via stdin to avoid having to write to a temporary file
+            logger.info(
+                "Loading authorization configuration from %s",
+                settings.AUTHORIZATIONS_CONFIG_FIXTURE_PATH,
+            )
+            with override_stdin(StringIO(content)):
+                try:
+                    call_command("loaddata", "-", format="yaml")
```

Comment:
My favorite question - should we have an additional validation for the fixture? For example, checking if zaaktypen are existing urls or checking that catalog autorisatie and individual autorisatie don't overlap? 

Maybe we can do it in the `test_configuration` part?

Reply:
I think validating it might be a good idea, though I'm not sure if `test_configuration` is the right place, since that runs after `configure`, so you'd be left with a selftestfailure, but the configuration was loaded.

I'm not sure if it's easy to keep all the validation between the admin interface, API and this configuration step in sync, I'll see what I can do with it

Reply:
I've been trying to do this, but I don't think it's possible to share the same validation between the admin form and the configuration step without duplicating it unfortunately, particularly validating the overlap between autorisaties/catalogusautorisaties is annoying, because the fixture has a different shape for the data than the AutorisatieFormSet (where multiple autorisaties are grouped together). To further complicate things, in the form validation `zaaktypen/iotypen/besluittypen` are resolved using django loose fk, while in the configurationstep this has to be done manually as well

@annashamray do you have any ideas? :grimacing:

Reply:
Let's park this for now and see how this configuration step is used to determine the next step